### PR TITLE
use vars to select eos_validate_state tests

### LINKED ISSUE #664: eos_validation_states - provide ability to select the tests using variables
**Issue Type**

- [ ] Template enhancement
- [x] Role enhancement
- [ ] Documentation enhancement (avd.sh)

**Is your feature request related to a problem? Please describe.**

eos_validation_states  runs many tests. we can use tags to select/skip them. it would be better to select the tests using variables.

### PR DESCRIPTION
**Pull request by ksator**
## Change Summary

eos_validation_state role enhancement. 
This role runs many tests. 
we were already able to use tags to select/skip them. 
Adding now vars to select the tests 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #664 

## Component(s) name

`arista.avd.eos_validate_state`

## Proposed changes
Added vars to select the tests in the `default` dir of the role. 
```
# Tests to include
eos_validate_state_bgp_check: true
eos_validate_state_hardware: true
eos_validate_state_interfaces_state: true
eos_validate_state_ip_reachability: true
eos_validate_state_lldp_topology: true
eos_validate_state_loopback0_reachability: true
eos_validate_state_mlag: true
eos_validate_state_ntp: true
eos_validate_state_routing_table: true
eos_validate_state_reload_cause: false
```
The tasks `main.yml` includes now a condition to load tests.  
 

## How to test

define in host_vars or group_vars or extra_vars how do we want to override the default: 
```
eos_validate_state_hardware: true
eos_validate_state_interfaces_state: true
eos_validate_state_ip_reachability: false
eos_validate_state_lldp_topology: false
eos_validate_state_loopback0_reachability: true
eos_validate_state_mlag: true
eos_validate_state_reload_cause: true
```
execute a playbook that has this role. 
and check the report output. 

```
ansible-playbook playbooks/playbook_validate_states.yml -i inventories/ebgp\ avd/inventory.yml
ansible-playbook playbooks/playbook_validate_states.yml -i inventories/ebgp\ avd/inventory.yml --skip mlag
```

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [ ] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_validate_state/tasks/main.yml [hunk 1]
```diff
@@ -73,59 +73,63 @@
...
+  when: eos_validate_state_interface_state is defined and eos_validate_state_interface_state is true
   tags:
     - interfaces_state
 
 - name: Load LLDP Topology tasks when dns domain is defined
   include_tasks: "lldp_topology_fqdn.yml"
   tags:
     - lldp_topology
-  when: dns_domain is defined
+  when: eos_validate_state_lldp_topology is defined and eos_validate_state_lldp_topology is true and dns_domain is defined
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_validate_state/tasks/main.yml:90]: As variables are defined in `default` section, you do not have to run this test `eos_validate_state_* is defined`

File: ansible_collections/arista/avd/roles/eos_validate_state/README.md [hunk 2]
```diff
@@ -85,6 +85,18 @@ root_dir: '{{ inventory_dir }}'
 eos_validate_state_name: 'reports'
 eos_validate_state_dir: '{{ root_dir }}/{{ eos_validate_state_name }}'
 
+# Tests to include
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_validate_state/README.md:88]: As it is variable description, we should put:

```
eos_validate_state_bgp_check: < true | false (Default: true) >
...
```

**Review by titom73 — CHANGES_REQUESTED** [review]
Nice PR, just few comments to address before we merge it.

File: ansible_collections/arista/avd/roles/eos_validate_state/tasks/main.yml (hunk 1 above)
Comment (ksator) [reply ansible_collections/arista/avd/roles/eos_validate_state/tasks/main.yml:90]: OK I will fix this

File: ansible_collections/arista/avd/roles/eos_validate_state/README.md (hunk 2 above)
Comment (ksator) [reply ansible_collections/arista/avd/roles/eos_validate_state/README.md:88]: OK I will fix this

**Comment by titom73** [comment]
Will be re-opened later

**Pull request closed without merging**